spec.bs

```diff
@@ -1356,6 +1467,204 @@ To <dfn>build an interest group passed to generateBid</dfn> given an [=interest
+        * |sig|["signature"] does not [=map/exist=], or it's not a [=string=].
+      1. Let |maybeKey| be the result of running [=forgiving-base64 decode=] with |sig|["key"].
+      1. Let |maybesignature| be the result of running [=forgiving-base64 decode=] with
+        |sig|["signature"].
+      1. Set |decodeSignatureFailed| to true and [=iteration/break=] if any of the following
+        conditions hold:
+        * |maybeKey| is failure, or its [=byte sequence/length=] is not 32;
+        * |maybesignature| is failure, or its [=byte sequence/length=] is not 64;
+      1. Let |signature| be a [=signed additional bid signatures=], whose
+        [=signed additional bid signature/key=] is |maybeKey|, and
+        [=signed additional bid signature/signature=] is |maybesignature|.
+      1. [=list/Append=] |signature| to |signatures|.
+    1. If |decodeSignatureFailed| is true, then decrement |auctionConfig|’s
+      [=auction config/scoring dependency count=], and [=iteration/continue=].
+    1. Let |decodedAdditionalBid| be the result of [=decode an additional bid=] with
+      |parsedSignedAdditionalBid|["bid"], |auctionConfig| and |topLevelAuctionConfig|.
+    1. Decrement |auctionConfig|’s [=auction config/scoring dependency count=], and
+      [=iteration/continue=] if any of the following conditions hold:
+      * |decodedAdditionalBid| is failure;
+      * |decodedAdditionalBid|'s [=additional bid/bid=]'s [=generated bid/bid=]'s
+        [=bid with currency/currency=] is not the result of running [=look up per-buyer currency=]
+        with |auctionConfig|.
+    1. Let |validSignatureIndices| be a new [=list=] of integers.
+    1. [=list/For each=] |i| in [=list/get the indices|indices=] of |signatures|:
+      1. If [=verify=] |signatures|[|i|] succeeded:
+        1. [=list/Append=] |i| to |validSignatureIndices|.
+    1. Let |negativeTargeted| be false.
+    1. Let |additionalBidBuyer| be |decodedAdditionalBid|'s [=additional bid/bid=]'s
+      [=generated bid/interest group=]'s [=interest group/owner=]. 
+    1. [=list/For each=] |igName| of |decodedAdditionalBid|'s
+      [=additional bid/negative target interest group names=]:
+      1. If |negativeInfo|[(|additionalBidBuyer|, |igName|)] [=map/exists=]:
+        1. Let (|joiningOrigin|, |additionalBidKey|) be |negativeInfo|[(|additionalBidBuyer|, |igName|)].
+        1. Let |KeyMatchedValidSignature| be false. 
+        1. [=list/For each=] |i| of |validSignatureIndices|:
+          1. If |signatures|[|i|]'s [=signed additional bid signature/key=] is |additionalBidKey|:
+            1. Set |KeyMatchedValidSignature| to true, and [=iteration/break=].
+        1. If |KeyMatchedValidSignature| is false, then [=iteration/continue=].
+        1. If |joiningOrigin| is not null:
+          1. If |joiningOrigin| is not [=same origin=] with |decodedAdditionalBid|'s
+            [=additional bid/negative target joining origin=], then [=iteration/continue=].

```

Comment:
Both this `[=iteration/continue=]` and the one above are really significant, in that they represent the "fail open" error scenarios that are important in both of these cases. Is there a way to add a comment here emphasizing that?

Reply:
We can add a "Note: " block, but not sure if we need to for this. I'll keep this open and ask our spec mentor's opinion when I ask them to review.

Reply:
When in doubt, I often add non-normative Notes to things, to be over-descriptive. So feel free to do that _if_ you think it'd be useful. (Make sure the indenting is right!)

Reply:
Added a note.